lib/tools/system-calls.js

```diff
@@ -396,10 +400,12 @@ systemCallMethods.shell = async function shell (cmd, opts = {}) {
   } finally {
     if (privileged && shouldRestoreUser && (!keepPrivileged || didCommandFail)) {
       try {
-        await this.adbExec(['unroot'], opts);
+        retryInterval(10, 1000, async () => await this.adbExec(['unroot'], opts));
```

Comment:
"blind" retry will only slow down a test. There should be a better way to distinguish the state after the command has been executed